packages/testwatch/index.js

```diff
@@ -220,7 +220,7 @@ ${Object.entries(this.#currentCommands)
   }
 
   async run() {
-    await this.#runTests();
+    this.#runTests();
     await once(this.#emitter, 'drained');
```

Comment:
```suggestion
    await Promise.race([this.#runTests(), once(this.#emitter, 'drained')]
```
I think this should be the solution - but it should probably be done in a separate PR?

Reply:
Great solution!
Why separate?

Reply:
ok. I pushed a revert for this change and will open another PR